native/spark-expr/src/structs.rs

```diff
@@ -136,7 +124,7 @@ impl PartialEq<dyn Any> for CreateNamedStruct {
                     .iter()
                     .zip(x.values.iter())
                     .all(|(a, b)| a.eq(b))
-                    && self.data_type.eq(&x.data_type)
+                    && self.names.iter().zip(x.names.iter()).all(|(a, b)| a.eq(b))
```

Comment:
do we also need a check that `self.names().len() == x.names().len()`?

Reply:
Good call. Added the checks.